src/main/java/com/bastagruppen/springskolsystem/model/Course.java

```diff
@@ -1,24 +1,86 @@
 package com.bastagruppen.springskolsystem.model;
 
-import jakarta.persistence.*;
-import lombok.*;
+import java.util.List;
+import java.util.UUID;
+
+import jakarta.persistence.Entity;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.PositiveOrZero;
+import jakarta.validation.constraints.Size;
+import static lombok.AccessLevel.PRIVATE;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.FieldDefaults;
 
 @Getter
 @Setter
-@AllArgsConstructor(access = AccessLevel.PUBLIC)
-@NoArgsConstructor(access = AccessLevel.PUBLIC)
+@FieldDefaults(level = PRIVATE)
+@AllArgsConstructor(access = PRIVATE)
 @Entity
-public class Course {
-    @Id
-    @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private long id;
-    @Column(nullable = false)
-    private String title;
-    @Column(nullable = false)
-    private String teacher;
-    @Column(nullable = false)
-    private int maxStudents;
-/*    @Setter(AccessLevel.NONE)
-    private final List<Student> students = new ArrayList<>(); // do we actually need this?*/
 
+public final class Course{
+
+    protected final UUID id = java.util.UUID.randomUUID();
```

Comment:
This has to be annotated with @Id and @GeneratedValue(strategy = GenerationType.UUID). No assignment to do here. Look at Student for en example.
Currently, it gives an exception when running